solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -557,7 +558,7 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
           while (rb.outgoing.size() == 0) {
             ShardResponse srsp =
                 tolerant
-                    ? shardHandler1.takeCompletedIncludingErrors()
+                    ? shardHandler1.takeCompletedIncludingErrorsWithTimeout(maxTimeAllowed)
```

Comment:
shouldn't it be the *remaining* time allowed?  
For example see:
https://github.com/apache/solr/blob/53a25d1b5d02dd234358264420884b8eed204b56/solr/core/src/java/org/apache/solr/search/TimeAllowedLimit.java#L53

Hmm; ideally it'd be a one-liner to get the remaining time allowed; maybe a utility method should be somewhere for this.

Reply:
Thanks @dsmiley ,  I have updated your feedback. Also @gus-asf please let me what you think about it